src/main/java/it/gov/pagopa/pu/debtpositions/service/massive/DebtPositionMassiveServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+      case InstallmentSynchronizeDTO.ActionEnum.A ->
+        workflowId = updateActionMassiveDebtPositionService.handleCancellation(debtPositionSynchronizeDTO, massive, accessToken);
+    }
+
+    return workflowId;
+  }
+
+  private DebtPositionTypeOrg retrieveDebtPositionTypeOrg(Long organizationId, String debtPositionTypeCode) {
+    return debtPositionTypeOrgRepository.findByOrganizationIdAndCode(
+        organizationId, debtPositionTypeCode)
+      .orElseThrow(() -> new InvalidValueException(String.format("The debt position type code %s is not valid for this organizationId %s", debtPositionTypeCode, organizationId)));
+  }
+
+  private void populateFirstTransfer(InstallmentSynchronizeDTO installmentSynchronizeDTO, String accessToken, DebtPositionTypeOrg debtPositionTypeOrg) {
+    Long organizationId = installmentSynchronizeDTO.getOrganizationId();
+
+    Organization organization = organizationSearchClient.findByOrganizationId(organizationId, accessToken);
+    String category = debtPositionTypeRepository.findByDebtPositionTypeId(debtPositionTypeOrg.getDebtPositionTypeId()).getTaxonomyCode();
+
+    BigDecimal totalAmountOtherTransfers = installmentSynchronizeDTO.getTransfersList().stream()
+      .filter(Objects::nonNull).map(TransferSynchronizeDTO::getAmount)
+      .filter(Objects::nonNull).reduce(BigDecimal.ZERO, BigDecimal::add);
+
+    TransferSynchronizeDTO firstTransfer = TransferSynchronizeDTO.builder()
+      .orgFiscalCode(organization.getOrgFiscalCode())
+      .orgName(organization.getOrgName())
+      .iban(debtPositionTypeOrg.getIban().isBlank() ? organization.getIban() : debtPositionTypeOrg.getIban())
+      .category(category)
+      .amount(installmentSynchronizeDTO.getAmount().min(totalAmountOtherTransfers))
+      .remittanceInformation(installmentSynchronizeDTO.getRemittanceInformation())
+      .build();
+
+    installmentSynchronizeDTO.getTransfersList().add(firstTransfer);
+  }
+
+  private DebtPositionDTO createDpFromInstallmentSynchronize(InstallmentSynchronizeDTO installmentSynchronizeDTO, String accessToken) {
+    DebtPositionTypeOrg debtPositionTypeOrg = retrieveDebtPositionTypeOrg(installmentSynchronizeDTO.getOrganizationId(),
+      installmentSynchronizeDTO.getDebtPositionTypeCode());
+
+    populateFirstTransfer(installmentSynchronizeDTO, accessToken, debtPositionTypeOrg);
+
+    if(installmentSynchronizeDTO.getBalance().isBlank()){
+      installmentSynchronizeDTO.setBalance(debtPositionTypeOrg.getBalance());
+    }
+
+    return installmentSynchronizeMapper.map2DebtPositionDTO(installmentSynchronizeDTO, debtPositionTypeOrg.getDebtPositionTypeOrgId());
+  }
+
+
+  private DebtPosition checkProcessableDebtPosition(String iupdOrg) {
+    Optional<DebtPosition> debtPosition = debtPositionRepository.findByIupdOrg(iupdOrg);
+
+    if (debtPosition.isPresent() && !DebtPositionOrigin.ORDINARY_SIL.equals(debtPosition.get().getDebtPositionOrigin())) {
```

Comment:
This could be simplified
```suggestion
    if (debtPosition.map(d -> !DebtPositionOrigin.ORDINARY_SIL.equals(d.getDebtPositionOrigin()).orElse(false)) {
```